dtfs-central-api/src/v1/controllers/portal/gef-deal/add-gef-deal-cancelled-activity.js

```diff
@@ -0,0 +1,36 @@
+const { getUnixTime } = require('date-fns');
+const { PORTAL_ACTIVITY_LABEL, PORTAL_ACTIVITY_TYPE } = require('@ukef/dtfs2-common');
+const { updateDeal } = require('./update-deal');
+
+/**
+ * Add a "deal cancelled" activity to a GEF deal.
+ * @param {string | ObjectId} addGefDealCancelledActivity.dealId - the dealId
+ * @param {import('@ukef/dtfs2-common').Activity[]} addGefDealCancelledActivity.portalActivities - previous/existing deal activities
+ * @param {import('@ukef/dtfs2-common').ActivityAuthor} addGefDealCancelledActivity.author - the activity's author
+ * @param {import('@ukef/dtfs2-common').AuditDetails} addGefDealCancelledActivity.auditDetails - the users audit details
+ * @returns {Promise<('@ukef/dtfs2-common').Deal>}
+ */
+exports.addGefDealCancelledActivity = async ({ dealId, portalActivities, author, auditDetails }) => {
```

Comment:
I tried to add unit tests for this, but it does not seem possible with the current patterns/approach.

Unless i'm missing something - spent a lot of time on this, It's not possible to mock certain things with the current JS approach. I tried transforming some things to TypeScript - got the tests working, but then there are type issues. Additionally, this'd be the only file in this area that is misaligned.

Would like to transform this whole area of the codebase to TS and use ES6 imports/exports. Then test coverage can be added.

Reply:
Resolving this as the funciton has since been deleted and we now have unit test coverage.